Automattic/woocommerce-payments
# Save tokens for Link payment method

Fixes #4703 

#### Changes proposed in this Pull Request
For subscriptions, the payment method token should be saved in order to be re-used for subscription renewal. 
This PR updates Link payment method to save the token.









#### Testing instructions





* **Scenario 1**
* Create a new subscription order and pay it using Link
* Payment should be completed
* The payment of renewal for the subscription should be processed without errors. To check this, use a 1 day subscription renewal.
* Subscription fee should be applied
* **Scenario 2**
* Create a new subscription with 1 day trial version order and pay it using Link
* Payment should be completed
* The payment for the subscription should be processed without errors after the trial ends.
* Subscription fee should be applied
* **Scenario 3**
* Install [Woocommerce Subscriptions](https://woocommerce.com/products/woocommerce-subscriptions/)
* When WCS is enabled it will take over the subscription functionality and will be managed by WCS.
* In order to trigger subscription renewals, you can follow the steps [ [here](https://woocommerce.com/document/testing-subscription-renewal-payments/) ]
* Repeat the tests above. 
* Subscription fee should NOT be applied.

Notes: 
1. In order to test the built-in subscription, you have to create a 1day subscription and wait for Stripe to renew the subscription. You cannot test it with Stripe's Subscription Clocks, because this will create a duplicate customer, and a new payment method ( credit card payment method ). 
2. If the subscription is not renewed after 1 day on client side, you can check the events on Stripe side for that subscription because you might have the listen webhook closed on your local when Stripe renewed the subscription.
3. [Subscriptions plugin](https://woocommerce.com/products/woocommerce-subscriptions/) overlaps the built in subscriptions.
4. In order to test the subscriptions plugin, please use the steps from [here](https://woocommerce.com/document/testing-subscription-renewal-payments/). 
-------------------


**Post merge**
